aptos-move/framework/aptos-framework/sources/aptos_governance.move

```diff
@@ -328,6 +328,15 @@ module aptos_framework::aptos_governance {
                 should_pass,
             },
         );
+
+        let proposal_state = voting::get_proposal_state<GovernanceProposal>(@aptos_framework, proposal_id);
+        if (proposal_state == PROPOSAL_STATE_SUCCEEDED) {
+            add_approved_script_hash(proposal_id);
+        }
+    }
+
+    public entry fun add_approved_script_hash_script(proposal_id: u64) acquires ApprovedExecutionHashes {
```

Comment:
why we need this entry function?

Reply:
We can't guarantee that a vote will immediately result in an approved. So we need to this as a fallback.